src/Analyzers/CSharp/Analyzers/RemoveUnreachableCode/CSharpRemoveUnreachableCodeDiagnosticAnalyzer.cs

```diff
@@ -57,7 +57,7 @@ private void AnalyzeSemanticModel(SemanticModelAnalysisContext context)
         // binding diagnostics directly on the SourceMethodSymbol containing this block, and
         // so it can retrieve the diagnostics at practically no cost.
         var root = semanticModel.SyntaxTree.GetRoot(cancellationToken);
-        var diagnostics = semanticModel.GetDiagnostics(context.FilterSpan, cancellationToken);
+        var diagnostics = semanticModel.GetMethodBodyDiagnostics(context.FilterSpan, cancellationToken);
```

Comment:
at the very least, i would doc this heavily as to why we're calling this, and what savings we're sure to achieve here.

Reply:
@CyrusNajmabadi I added a comment. Is it good to go now?